### ISSUE TITLE
Docker cannot connect to Repository Path repository

### ISSUE THREAD

**Issue by stieler-it** [jfrog/charts#87]
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:

Helm (but we are using Rancher to deploy Artifactory, not Helm directly):
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}

Kubernetes:
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-17T18:43:26Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}

**Which chart**: artifactory

**What happened**: Error when trying to `docker login`.

**What you expected to happen**: Artifactory can be used as a Docker repository in "Repository Path" mode without further (proxy) configuration.

**How to reproduce it** (as minimally and precisely as possible):

1. Deploy Artifactory using official Helm chart with ingress.enabled = true
2. Configure Docker repositories
3. Use "Repository path" method
4. Create a user to access
5. Execute docker login <artifactory-url>
6. Enter credentials

**Anything else we need to know**:

The problem only shows up when accessing Artifactory through Ingress / nginx. Then, we cannot login to Docker repository:

```
[rancher@intern-worker-lg3 ~]$ docker login artifactory.our.hostname
Username: docker
Password:
Error response from daemon: Get https://artifactory.our.hostname/v2/: unable to decode token response: EOF
```

Artifactory requests log also looks wrong as the call to `/v2/token` shown up as `/v2` only:

```
20181015205810|1|REQUEST|10.0.0.22|non_authenticated_user|GET|/api/docker/v2/|HTTP/1.1|401|0
20181015205810|1|REQUEST|10.0.0.22|non_authenticated_user|GET|/api/docker/v2/|HTTP/1.1|200|0
```

It does work when bypassing the Ingress/nginx reverse proxy by connecting to the K8s service directly:

```
[rancher@intern-worker-lg3 ~]$ docker login 10.43.245.110:8081
Username: docker
Password:
Login Succeeded
```

Artifactory request logs look better:
```
20181015205657|1|REQUEST|10.0.0.10|non_authenticated_user|GET|/api/docker/v2/|HTTP/1.1|401|0
20181015205657|237|REQUEST|10.0.0.22|k8sworker|GET|/api/docker/null/v2/token|HTTP/1.1|200|0
20181015205657|9|REQUEST|10.0.0.10|k8sworker|GET|/api/docker/v2/|HTTP/1.1|200|0
```

There is a workaround:

In a session with JFrog support team, we tried to set some proxy headers in Nginx proxy, which didn't help. We could solve it by adding a rewrite rule to the reverse proxy: `rewrite ^/v2/token$ /artifactory/api/docker/null/v2/token;`. This can be applied by setting the Helm chart value `ingress.annotations.nginx\.ingress\.kubernetes\.io/configuration-snippet`.

**Comment by jainishshah17** [comment]
@stieler-it have you tried following this: https://medium.com/@jainishshah17/kubernetes-nginx-ingress-with-artifactory-on-gke-e6781d4ab5e8

**Comment by stieler-it** [comment]
Thank you. However, you seem to be using even more workarounds than we are. Don't you think that these Ingress customization rules should be part of the Helm chart to work out of the box?

**Comment by jainishshah17** [comment]
https://github.com/jfrog/charts/pull/91

**Comment by eldada** [comment]
@stieler-it - did you try the suggested fix in #91 ?

**Comment by stieler-it** [comment]
I have not tried it because I currently don't have an environment to play with. However, the important change is in the documentation and this together with the other modifications look good to me + like exactly what we need.
